### PR TITLE
Invert patch-based transforms

### PR DESCRIPTION
**Pull request by rijobro**
Adds inverse for:
- RandSpatialCropSamplesd
- RandWeightedCropd
- RandCropByPosNegLabeld

Fixes https://github.com/Project-MONAI/MONAI/issues/1901. 
cc @diazandr3s.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/croppad/dictionary.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 Class names are ended with 'd' to denote dictionary-based transforms.
 """
 
+from contextlib import nullcontext
```
Comment (wyli) [inline monai/transforms/croppad/dictionary.py:18]: this module is not in python 3.6...

Comment (rijobro) [reply monai/transforms/croppad/dictionary.py]: Thanks, I've modified to not use it.

**Review by wyli — APPROVED** [review]
thanks!

**Comment by rijobro** [comment]
@Nic-Ma do you think this will this cause any problems with your current workflows?

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
Could you please help add a comment or doc-string to highlight that: we can't automatically `batch-invert` the entire transform chain if contains patch-based transform, because the output will be a list?

Thanks.

**Pull request merged**